include/mc_control/ControllerClient.h

```diff
@@ -590,12 +590,15 @@ struct MC_CONTROL_CLIENT_DLLAPI ControllerClient
    *
    * \p send_index If true, the implementation should send back the index
    * rather than the value
+   *
+   * \p def default index
    */
   virtual void form_combo_input(const ElementId & /*formId*/,
                                 const std::string & /*name*/,
                                 bool /*required*/,
                                 const std::vector<std::string> & /*values*/,
-                                bool /*send_index*/)
+                                bool /*send_index*/,
+                                int /*def*/)
   {
   }
```

Comment:
Sorry if that part was not clear, to retain API compatibility, we need:


```cpp
      /** A combo input within a form
     *
     * This is kept for backward compatibility, mc_rtc never calls this version and you should implement the full version
     */
    virtual void form_combo_input(const ElementId & /*formId*/,
                                  const std::string & /*name*/,
                                  bool /*required*/,
                                  const std::vector<std::string> & /*values*/,
                                  bool /*send_index*/)
  {
  }

    /** A combo input within a form
     *
     * \p formId Identifier of the form
     *
     * \p name Name of the entry
     *
     * \p required If true, it must hold a value when the form is sent
     *
     * \p values Possible values
     *
     * \p send_index If true, the implementation should send back the index
     * rather than the value
     */
    virtual void form_combo_input(const ElementId & formId,
                                  const std::string & name,
                                  bool required,
                                  const std::vector<std::string> & values,
                                  bool send_index,
                                  int /*def*/)
    {
      form_combo_input(formId, name, required, values, send_index);
    }
```

Reply:
fixed in https://github.com/jrl-umi3218/mc_rtc/pull/178/commits/2f5054bdc0a6757e6ab6f0a6586fa435ad37cbaf